### PR TITLE
Template builder

### PR DESCRIPTION
**Pull request by andreievg**
Required for front end branch in this PR: https://github.com/openmsupply/application-manager-web-app/pull/841

* Template section to be unique code within a template
* Template version timestamp to be updated when inserted/duplicated
* Template version trigger is also running on update (so that if template code is change on a template to an existing code, it doesn't cause conflict in version)
* Add isAdmin to permission policy, for UI control of admin options
* Extend template element and template action to get parameters as string (so we can query for text existing in the parameters, for permission policy info)
* Application to have is_config field to differentiate applications created in template builder vs normal applications
* Helper to delete whole application (including related records)
* Expose pg_policies view, for permission name info
* Change how permission row policies are created and configured (will make a seperate issue and PR for this), moral of the story, it was too slow to have row policy for each templatePermission (especially when permission restrictions are 'nested' and rely on other tables), so made row policies per policy only.
* Transferred lao users and org config from demo branch
* Snapshots and import from json to work for template export, import and duplication

### CODE REVIEW

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Nice work! I don't pretend to understand everything that's going on, but it's an impressive achievement. You'll have to explain to us the changes you made to the permissions at some point.

Couple of very minor comments.

File: database/buildSchema/16_application.sql [hunk 1]
```diff
@@ -11,6 +11,7 @@ CREATE TABLE public.application (
     user_id integer REFERENCES public.user (id),
     org_id integer REFERENCES public.organisation (id),
     session_id varchar,
+    is_config bool DEFAULT FALSE,
```
[the remark this answers is not in the document]
Comment (andreievg) [reply database/buildSchema/16_application.sql]: c252d9e6ae57af4683d1e7fedbf46b2d221bcf7e

File: src/components/snapshots/takeSnapshot.ts [hunk 2]
```diff
@@ -50,7 +51,7 @@ const takeSnapshot: SnapshotOperation = async ({
       JSON.stringify(options, null, ' ')
     )
 
-    await getSchemaDiff(newSnapshotFolder)
+    if (options.shouldReInitilise) await getSchemaDiff(newSnapshotFolder)
```
[the remark this answers is not in the document]
Comment (andreievg) [reply src/components/snapshots/takeSnapshot.ts]: c252d9e6ae57af4683d1e7fedbf46b2d221bcf7e

File: database/snapshotOptions/templateExport.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "templateSection",
+    "templateStageReviewLevel",
+    "permissionName",
+    "permissionPoliy"
+  ],
+  "tablesToUpdateOnInsertFail": ["permissionName", "permissionPolicy", "filter"],
+  "excludeTables": [],
+  "insertScriptsLocale": "dev",
+  "includeInsertScripts": [],
+  "excludeInsertScripts": []
```
[the remark this answers is not in the document]
Comment (andreievg) [reply database/snapshotOptions/templateExport.json:21]: work in progress

**Review by CarlosNZ — APPROVED** [review]

**Pull request merged**